ra2ce/analysis/losses/losses_base.py

```diff
@@ -94,7 +72,7 @@ def __init__(
 
         self.link_id = analysis_config.config_data.network.file_id
         self.link_type_column = analysis_config.config_data.network.link_type_column
-        self.trip_purposes = self.analysis.trip_purposes
+        self.trip_purposes: list[TripPurposeEnum] = self.analysis.trip_purposes
```

Comment:
why is `analysis` still needed? Ideally we map it to explicit properties like you did for `trip_purposes` can it be removed already?

Reply:
`self.analysis` contains the configuration of the analysis section from the `analysis_config.config_data.analyses` that is currently executed. So, I do think it is indeed needed still.

Reply:
Discussed: as we're in the init still it is fine like this